contracts/Core/StakeManager.sol

```diff
@@ -111,12 +110,12 @@ contract StakeManager is Initializable, ACL, StakeStorage, StateManager, Pause {
         uint256 totalSupply = sToken.totalSupply();
         uint256 toMint = _convertRZRtoSRZR(amount, stakers[stakerId].stake, totalSupply);
 
-        // Step 3: Increase given stakers stake by : Amount
-        stakers[stakerId].stake = stakers[stakerId].stake + (amount);
-
-        // Step 4:  Mint sToken as Amount * (totalSupplyOfToken/previousStake)
+        // Step 3:  Mint sToken as Amount * (totalSupplyOfToken/previousStake)
         sToken.mint(msg.sender, toMint);
 
+        // Step 4: Increase given stakers stake by : Amount
+        stakers[stakerId].stake = stakers[stakerId].stake + (amount);
+
         emit Delegated(msg.sender, epoch, stakerId, stakers[stakerId].stake, block.timestamp);
```

Comment:
needs to be moved up

Reply:
I think this would be solved once we merge adtiyas PR

Reply:
oh no we can't move this up 
mint has to happen before stake increase as there is dependency

when we mint, in beforeTokenTransfer, we calculate razor deposited 
their we need an old stake, not the new one 

https://github.com/abhishekvispute/contracts/blob/52ec788e54fa8d320f0863944173485675948aa9/contracts/tokenization/StakedToken.sol#L63

Reply:
resolved 
As we discussed modified mint directly. 
https://github.com/abhishekvispute/contracts/blob/74bb27c6e26dd2c66a203e9469ec3b81531eae95/contracts/tokenization/StakedToken.sol#L37